pandas/core/frame.py

```diff
@@ -2212,6 +2212,22 @@ def insert(self, loc, column, value, allow_duplicates=False):
         self._data.insert(
             loc, column, value, allow_duplicates=allow_duplicates)
 
+    def transform(self, **kwargs):
+        """
+
+        """
+        data = self.copy()
```

Comment:
I misunderstand pandas's datamodel, presuming that you could do a shallow copy and then update a column without modifying the original (like a dict). I see now that I was mistaken :(.